src/checker.rs

```diff
@@ -49,7 +50,50 @@ mod tests {
         let profile = "release";
 
         let expected = format!("{}/target/{}/deps", pwd, profile);
-        assert_eq!(expected, get_base_path().unwrap());
+        let exe = env::current_exe();
+
+        match exe {
+            Ok(path) => assert_eq!(expected, get_base_path(path).unwrap()),
+            Err(_) => assert_eq!(true, false),
+        };
+    }
+
+    #[test]
+    fn test_download_and_unpack() -> Result<()> {
+        let base_url = generate_base_url("2.0.3");
+        let exe = env::current_exe();
+        let mut base_path = String::from("");
+        match exe {
+            Ok(path) => base_path = get_base_path(path).unwrap(),
+            Err(_) => assert_eq!(true, false),
+        }
+
+        let filename = generate_filename(
+            &get_os_type(sys_info::os_type()?).unwrap(),
+            get_architecture().unwrap(),
+        );
+
+        let result = download(&base_url, &base_path, &filename);
+        let tar_path = format!("{}/{}.tar.gz", base_path, filename);
+        assert_eq!((), result.unwrap());
```

Comment:
`assert!(result.is_ok())` might communicate the intention of the test a bit better.